Fixes to `#spriteOn:indent:`

janvrany opened the pull request:
This PR contains some fixes to Sprite printing introduces in PR #582

janvrany committed:
Add missing `SpriteRDecl >> isRecursive`

janvrany committed:
Fix printing "let in let"

This prints "let in let" as

    let a = {
      let b = 1;
      b + 1;
    };

This is in sync with what `ΛκParser` parses.

shingarov commented:
I absolutely hate the style of replacing Smalltalk-80's ad-hoc polymorphism with Algol's `if` by dispatching through "testers".  Smalltalk-80's Booleans are among the few Smalltalk concepts that Smalltalk-80 actually implements correctly.  So, the Smalltalk-80 already dispatches on the class of the receiver.  Why "monospatch" it, moving the point of decision-making to Boolean??

janvrany committed:
Fix printing of binary operators

While binary operators like `==` or `>` are technically applications,
on syntactic level they're infix operators. This commit fixes printing
so they're printed as such.

janvrany commented:
Are you talking about aebdf11a? If so, do you like 9543ee9 more? Or shall it go all the way down to `PrimOp`?
